html_css/v2/foundations/css-foundations.md

```diff
@@ -0,0 +1,425 @@
+
+When it comes to internal and external CSS, each type of selector has its own specificity level (there are other selectors that make up the following list, but we're focusing on the ones mentioned in this lesson):
+
+1. ID selectors (most specific, after inline CSS)
+2. Class selectors
+3. Type selectors
+4. Universal selector and combinators (no specificity)
+
+Specificity will only be taken into account when an element has multiple rules targeting it, sort of like a tie-breaker. Ignoring universal selectors and combinators, the other types of selectors will increase the specificity of a rule for that selector's level.
+
+Specificity is represented by a comma-separated list of numbers: `0, 0, 0`. Starting from the left, each number represents the three specificity levels (id, class, and type), with each number having a higher precedence than the next (ID beats all, class beats type and no specificity, type beats anything left over). 
+
+Think of specificity as a sort of scoreboard. If an ID selector is used, the "score" becomes `1, 0, 0`, whereas if a type selector is used the "score" becomes `0, 0, 1`. We would then compare the specificity of each rule to see which one wins; in this case the ID selector would win because `1, 0, 0` is a higher "score" than `0, 0, 1` (ID beats type).
+
+Take a look at the example below, and see if you can determine which rule would be applied:
+
+~~~css
+/* rule 1 */
+#main {
+  color: blue;
+}
+
+/* rule 2 */
+.subsection .list {
+  color: red;
+}
+~~~
+
+Do you have an answer ready? Are you sure? Okay. In this example, rule 1 only has an ID selector, so its specificity is `1, 0, 0`, while rule 2 has a specificity of `0, 2, 0` because it has two class selectors. In this case, rule 1 would win. Another way to look at this list of numbers is just as a normal number: 100 is a higher number than 020.
+
+Quick note: you can actually view this specificity in VS Code when you hover over a selector in a CSS file!
+
+#### Inheritance
+
+There are some CSS properties that, when applied to an element, will be inherited by descendants of that element, while other properties don't get inherited. An example of a property that *is* inherited is the `color` property. If `color: red` were applied to an element, any descendants of that element would inherit that same `color` property.
+
+The exception to this is when directly targeting an element, as this always beats inheritance:
+
+~~~html
+<!-- index.html -->
+
+<div id="parent">
+  <div class="child"></div>
+</div>
+~~~
+~~~css
+/* styles.css */
+
+#parent {
+  color: red;
+}
+
+.child {
+  color: blue
+}
+~~~
+
+Despite the `parent` element having a higher specificity with an ID, the `child` element would have the `color: blue` style applied since that declaration directly targets it, while `color: red` from the parent is only inherited.
```

Comment:
I'd add a note that says something like 
> typography related properties (font, color, letter-spacing etc.) usually inherit, but most other things do not.

Just so people have some idea of what to expect when it comes to inheritance.